xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -31,15 +31,15 @@ using namespace KODI::MESSAGING;
 
 namespace
 {
-  int ClientIdFromAddonId(const std::string& strID)
-  {
-    std::hash<std::string> hasher;
-    int iClientId = static_cast<int>(hasher(strID));
-    if (iClientId < 0)
-      iClientId = -iClientId;
-    return iClientId;
-  }
-
+int ClientIdFromAddonId(const std::string& addonID, uint32_t instance)
+{
+  std::hash<std::string> hasher;
+  int iClientId =
+      static_cast<int>(hasher(addonID + (instance > 0 ? "@" + std::to_string(instance) : "")));
```

Comment:
see my other comment: syntax should be changed either to "instance@addon" or "addon::instance" - current syntax implies to read this as addon "at" instance, which feels like  wrong order